File: go/neetcode150/contains_duplicate.go
package neetcode150

// https://leetcode.com/problems/contains-duplicate/
func containsDuplicate(nums []int) bool {
	for i := 0; i < len(nums)-1; i++ {
		if nums[i] == nums[i+1] {
			return true
		}
	}
	table := make(map[int]int)
	for _, num := range nums {
		if table[num] != 0 {
			return true
		} else {
			table[num] = 1
		}
	}
	return false
}

File: src/main/java/neetcode150/array/ContainsDuplicate.java
package neetcode150.array;

import java.util.HashSet;

// https://leetcode.com/problems/contains-duplicate/
public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        for(int i=0;i<nums.length-1;i++)
        {
            if(nums[i]==nums[i+1])
                return true;
        }
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (set.contains(num)) {
                return true;
            }
            set.add(num);
        }
        return false;
    }
}